### ISSUE TITLE
Lunch scraper no longer works

### ISSUE THREAD

**Issue by Chaldron** [stevenson-space/shs#79]
Seems like the lunch scraper is broken, since the official website has switched from a scrapable HTML menu to a PDF menu [linked on this page.](https://www.d125.org/student-life/food-services).

We'll probably need to crawl the page to find the latest PDF link, and do some PDF parsing to reconstruct the menu. Hopefully they'll keep the menu structure the same every month.

**Comment by abhinavpappu** [comment]
Oof that sucks, PDFs are really hard to parse, especially if it's structured similarly to the PDFs they used 5 years ago :(

I'm just going to disable the lunch parser for now and hopefully the 28-day cycle we already have is accurate enough for a while

**Comment by abhinavpappu** [comment]
Wait I think the lunches are already inaccurate, going to hide the card for now and look at it later

**Comment by jshepin** [comment]
Good catch. Today was the first day they were off. We could use AWS textract for this. If they don't change the menu items, I suppose we could keep the current Json file and manually set a start point.

**Comment by jshepin** [comment]
<img width="1012" alt="Screen Shot 2021-09-01 at 8 54 21 PM" src="https://user-images.githubusercontent.com/31457361/131768838-9af666dd-0bc1-451b-a676-9d85a8ea9783.png">
the api using the "tables" function seems to pick it up pretty well

**Comment by Chaldron** [comment]
Hm, that is really interesting. I would prefer a tool we can use locally (and in the Netlify build runner) without having to interface with an external cloud service. Then we'd have to setup the credentials, make sure we don't go beyond the free tier limits, plus it's another thing to deal with when transitioning between people who are working on the site. If we can't find an alternative, then I wouldn't mind using this though. It seems to do a pretty good job. 

There are also Python/JS PDF parsing libraries - sometimes, if the PDF is "well-made", its structured in a way where it's really easy to extract the information. I would look into that as well.

**Comment by jshepin** [comment]
yeah that's understandable and actually not needed. The PDF has selectable text which those libraries should be able to use.

**Comment by jshepin** [comment]
I'm attempting to use pdf-parser to solve this issue. 4998615d019ab3a63dfce253de04fbd1a11da52a

This is the second time in the past 7 months they have changed their lunch menu site. Hopefully this one is here to stay 🙏🏻

**Comment by jshepin** [comment]
I worked on parsing the pdf, but currently, it cannot accurately ignore the other text on the page, and getting the dates to line up doesn't work well. Instead of reading the whole pdf line by line, what we could do is chop up the pdf into a grid of individual, smaller PDFs and read each of those. That way we can easily ignore the misc text on the page.

for a temporary solution for the month of september, Please check GH-80. Basically how it works, is that it takes the output form AWS textract and formats it to the `lunch.json`. Doing this buys us more time to work on parsing the pdf.

The solution can be extended beyond the month of october by manually running the next month's PDF through the api and changing the hardcoded variable.
